### ISSUE TITLE
I need to store a vector of vector<std::string> as a field in a persistent class, how can I  do this?

### ISSUE THREAD

**Issue by juandent** [fnc12/sqlite_orm#1255]
```C++
 class X  {
		using Row = std::vector<std::string>;
		using StringMatrix = std::vector<Row>;

		StringMatrix		m_matrix;
   }
```

**Comment by fnc12** [comment]
how would you store this in raw SQLite?

**Comment by juandent** [comment]
I don't know

**Comment by fnc12** [comment]
I don't think so. How would you solve this issue if you don't have `sqlite_orm` but only sqlite3 C library?

**Comment by juandent** [comment]
what is wrong with this proposal?

**Comment by fnc12** [comment]
Nothing wrong. But to answer this in terms of `sqlite_orm` one has to answer this in terms of SQLite first. `sqlite_orm` doesn't think instead of developer it just makes using SQLite easier. If the developer doesn't know how (s)he wants to store complex structure (vector of vectors in your case) in SQLite column no ORM can help

**Comment by juandent** [comment]
can you suggest what tables/relationships  could work?

**Comment by juandent** [comment]
I am stuck here

**Comment by fnc12** [comment]
there are plenty of options and it depends on different conditions including developer's personal preferences. E.g. https://stackoverflow.com/questions/3005231/how-to-store-array-in-one-column-in-sqlite3 . Globally we have two ways:
1) move arrays and subarrays into dedicated table(s) like it is done in subentities example in our repo
2) encode/decode the whole array structure into single dynamic value (like blob or string) e.g. json, xml, invent your own an after that adopt this (en/de)coding into custom types binding just like it is done in enum binding examples in our repo
